homeassistant/config.py

```diff
@@ -603,7 +604,18 @@ async def async_process_ha_core_config(hass: HomeAssistant, config: dict) -> Non
     hass.data[DATA_CUSTOMIZE] = EntityValues(cust_exact, cust_domain, cust_glob)
 
     if CONF_UNIT_SYSTEM in config:
-        if config[CONF_UNIT_SYSTEM] == CONF_UNIT_SYSTEM_IMPERIAL:
+        if config[CONF_UNIT_SYSTEM] == CONF_UNIT_SYSTEM_US_CUSTOMARY:
+            hac.units = US_CUSTOMARY_SYSTEM
+        elif config[CONF_UNIT_SYSTEM] == CONF_UNIT_SYSTEM_IMPERIAL:
+            _LOGGER.warning(
+                "Found deprecated unit system in core "
+                "configuration expected unit system. Replace '%s: %s' "
+                "with '%s: %s'",
+                CONF_UNIT_SYSTEM,
+                CONF_UNIT_SYSTEM_IMPERIAL,
+                CONF_UNIT_SYSTEM,
+                CONF_UNIT_SYSTEM_US_CUSTOMARY,
+            )
```

Comment:
Why do we put this burden on the user?

We cannot deprecate "imperial" as it is a valid unit system. We should come up with a migration path and plan (and migrate to a new configuration option) to solve this issue.

Reply:
After some discussions, I think we arrived at:
- Keep using the `unit_system` configuration key
- Remove the `imperial` setting from the documentation
- If we add a British imperial unit system in the future, we'll name it `british_imperial`

Reply:
Since we are looking to change the name of the unit system, I think that #80257 and linked PRs should be merged first.